lib/logstash/filters/hashid.rb

```diff
@@ -0,0 +1,121 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+require "openssl"
+
+# This filter allow you to generate predictable, string encoded hashed keys 
+# based om event contents and timestamp. This can be used to avoid getting 
+# duplicate records indexed into Elasticsearch.
+#
+# Hashed keys to be generated based on full or partial hashes and
+# has the ability to prefix these keys based on the event timestamp in order
+# to make then largely ordered by timestamp, which tend to lead to increased
+# indexing performance for event based use cases where data is being indexed
+# in near real time.
+#
+# When used with the timestamp prefix enabled, it should ideally be run after 
+# the date filter has run and populated the @timestamp field.
+class LogStash::Filters::Hashid < LogStash::Filters::Base
+  config_name "hashid"
+
+  # Source field(s) to base the hash calculation on
+  config :source, :validate => :array, :default => ['message']
+
+  # Timestamp field to use for the timestamp prefix
+  config :timestamp_field, :validate => :string, :default => '@timestamp'
+
+  # Target field.
+  # Will overwrite current value of a field if it exists.
+  config :target, :validate => :string, :default => 'hashid'
+
+  # Encryption key to be used when generating cryptographic hashes
+  config :key, :validate => :string, :default => 'hashid'
+
+  # Hash function to use
+  config :method, :validate => ['SHA1', 'SHA256', 'SHA384', 'SHA512', 'MD5'], :default => 'MD5'
+
+  # If full hash generated is not to be used, this parameter specifies how many bytes that should be used
+  # If not specified, the full hash will be used
+  config :hash_bytes_used, :validate => :number
+
+  # Use the timestamp to generate an ID prefix
+  config :timestamp_prefix, :validate => :boolean, :default => true
```

Comment:
Consider changing this to `prepend_timestamp` or `add_timestamp_prefix` as it reads like a directive to modify the timestamp ATM.